drivers/flash/CMakeLists.txt

```diff
@@ -22,13 +22,16 @@ zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_RV32M1 soc_flash_rv32m1.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_STM32_QSPI flash_stm32_qspi.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G flash_mcux_flexspi_mx25um51345g.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_nor.c)
+zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_HYPERFLASH flash_mcux_flexspi_hyperflash.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_ESP32 flash_esp32.c)
 
 if(CONFIG_FLASH_MCUX_FLEXSPI_XIP)
   if(CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G)
 	  zephyr_code_relocate(flash_mcux_flexspi_mx25um51345g.c ${CONFIG_FLASH_MCUX_FLEXSPI_XIP_MEM}_TEXT)
-  else()
+  elseif(CONFIG_FLASH_MCUX_FLEXSPI_NOR)
 	  zephyr_code_relocate(flash_mcux_flexspi_nor.c ${CONFIG_FLASH_MCUX_FLEXSPI_XIP_MEM}_TEXT)
+  elseif(CONFIG_FLASH_MCUX_FLEXSPI_HYPERFLASH)
+    zephyr_code_relocate(flash_mcux_flexspi_hyperflash.c ${CONFIG_FLASH_MCUX_FLEXSPI_XIP_MEM}_TEXT)
   endif()
```

Comment:
I would assume that this was actually not made properly in the first place since you could choose both NOR driver and MX25UM51345G driver. I will give it a go to fix it.

Reply:
yep, this is what unfortunately happen sometimes, and the next person adding things end up having to cleanup the mess of the past.